### PR TITLE
Member Dropdown Anim

### PR DESCRIPTION
**Pull request by eldiegod**
fixes: #1763

https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=3201%3A111989

### CODE REVIEW

**Review by rafalpawlow — CHANGES_REQUESTED** [review]
It's impossible to click **Upload video** button because dropdown container is hovering it
Zrzut ekranu 2021-12-22 o 01 05 22

File: src/components/_overlays/MemberDropdown/MemberDropdown.styles.ts [hunk 1]
```diff
@@ -9,27 +10,38 @@ const paddingStyles = css`
...
+  width: 280px;
+  position: 'relative';
+  max-height: calc(100vh - ${sizes(4)} - var(--size-topbar-height));
+
+  height: ${({ containerHeight }) => containerHeight}px;
+  transform: translateY(
+    ${({ isActive, containerHeight }) => (isActive ? 'var(--size-topbar-height)' : -containerHeight + 'px')}
+  );
+  transition: transform ${transitions.timings.loading} ${transitions.easing},
+    height ${cVar('animationTimingMedium')} ${cVar('animationEasingMedium')};
```
Comment (drillprop) [inline src/components/_overlays/MemberDropdown/MemberDropdown.styles.ts]: You can just write: 
```suggestion
    height ${cVar('animationTransitionMedium')};
```
Also, I think we should avoid using `{transitions.timings.loading} ${transitions.easing},` it should be deprecated soon. Could we change it to `animationTransitionSlow` ? Is there a reason that we used these values here?

File: src/components/_overlays/MemberDropdown/MemberDropdown.tsx [hunk 2]
```diff
@@ -97,119 +111,132 @@ export const MemberDropdown = React.forwardRef<HTMLDivElement, MemberDropdownPro
       }
     }
 
+    useEffect(() => {
+      transRef.start()
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (drillprop) [inline src/components/_overlays/MemberDropdown/MemberDropdown.tsx]: Why? I think we could safely have just: 
```tsx
    useEffect(() => {
      transRef.start()
    }, [isSwitchingMember, transRef])
```
Tried that myself and it seems that everything works as it should

**Review by drillprop — CHANGES_REQUESTED** [review]
Generally looks good, but found some issues. 
If you refresh the page you'll see the dropdown animation:

https://user-images.githubusercontent.com/51168865/147207984-f7786d07-4dc3-4918-9e3c-55ce8abf9526.mov

 It's even worse on playground:

https://user-images.githubusercontent.com/51168865/147208130-226491f9-3a68-4880-8365-c779f390896d.mov

**Comment by toiletgranny** [comment]
@eldiegod I can't seem to be able to preview the dropdown. Whenever I open up the Vercel preview link I get the "Oops! An error occurred." screen on Atlas.

**Comment by eldiegod** [comment]
> @eldiegod I can't seem to be able to preview the dropdown. Whenever I open up the Vercel preview link I get the "Oops! An error occurred." screen on Atlas.

@toiletgranny you should be able to review now, lmk!

**Comment by eldiegod** [comment]
> Generally looks good, but found some issues. If you refresh the page you'll see the dropdown animation:
> Nagranie.z.ekranu.2021-12-23.o.08.48.08.mov
> 
> It's even worse on playground:
> Nagranie.z.ekranu.2021-12-23.o.09.00.13.mov

@drillprop fixed the dropdown appearing briefly after refreshing, also don't mind too much about the playground start anim

**Comment by toiletgranny** [comment]
Looks awesome! Here are all the things I've found out: https://www.loom.com/share/bb73cda149f8496394eb5964ab6583b9

I'm aware that some of the things I'm bringing up in the above video might need to be tackled in separate issues, but I didn't want to end up recording and sharing multiple videos. Hope that's okay!

**Comment by drillprop** [comment]
> Looks awesome! Here are all the things I've found out: https://www.loom.com/share/bb73cda149f8496394eb5964ab6583b9
> 
> I'm aware that some of the things I'm bringing up in the above video might need to be tackled in separate issues, but I didn't want to end up recording and sharing multiple videos. Hope that's okay!

Just FYI the issues which you mentioned at the last minute of the video(switching members and channels page) should be already fixed in https://github.com/Joystream/atlas/pull/1788
@eldiegod you shouldn't bother about that much

**Pull request merged**